plutus-use-cases/test/Spec/Crowdfunding.hs

```diff
@@ -84,13 +76,7 @@ successfulCampaign = checkCFTrace scenario1 $ do
     con2 <- contrib2 c 600
     con3 <- contrib3 c 800
     updateAll'
-    setValidationData $ ValidationData $(plutus [| let el = []::[Signature] in
-            $(pendingTxCrowdfunding)
-                11
-                (Signature 1:el)
-                (600, Signature 2 : el)
-                (Just (800, Signature 3 : el))
-            |])
+    addBlocks 10
```

Comment:
why is this added?

Reply:
To ensure that enough "time" (blocks) has passed so that we can retrieve the funds or claim a refund. 

Block height is passed to the validation script as part of the `PendingTx` value. Previously we had to encode the desired `PendingTx` value in TH in order to compile it with the plugin (so the block height of the mockchain was different from what we passed to the validator script). Now the emulator uses the actual mockchain height, which (at this stage in the `Trace`) is no large enough for any of the conditions in the validator script to become true. So we have to add a number of empty blocks to simulate the passing of time.

Reply:
Perhaps you should add some comments to your mock traces explaining what they're doing :)

Reply:
Good idea...